### ISSUE TITLE
[JOSS review] API a bit cluncky

### ISSUE THREAD

**Issue by elena-pascal** [FrederikLizakJohansen/DebyeCalculator#28]
This is part of the review feedback for JOSS submission (openjournals/joss-reviews#6024)

From the usage example and available documentation it is not clear what is the benefit of passing the file at the method level and not at the object level together with the rest of the variables. It feel a bit clunky to pass the actual data after all the computation parameters have been initialised and computed.

**Comment by AndySAnker** [comment]
Thank you for the feedback. We acknowledge that it could be codewise elegant to load the structure during the initialisation of the object to prepare all calculations at the object level. In that case, 'DebyeCalculator' would be used in the following way:

```python
# Define structure sources
xyz_file = "some_path/some_file.xyz"
structure_tuple = (
    ["Fe", "Fe", "O", "O"],
    torch.tensor(
        [[0.5377, 0.7068, 0.8589],
         [0.1576, 0.1456, 0.8799],
         [0.5932, 0.0204, 0.6759],
         [0.6946, 0.4114, 0.4869]]
))

# Initialise calculator object
calc = DebyeCalculator(qmin=1.0, qmax=8.0, qstep=0.01, structure_source=xyz_file/structure_tuple)

# Calculate Powder (X-ray) Diffraction
Q, I = calc.iq()

# Update structure file
calc.update_structure(structure_source=xyz_file/structure_tuple)

# Calculate Powder (X-ray) Diffraction
Q, I = calc.iq()
```

Our implementation is inspired by 1) the implementation in DiffPy-CMI (see https://github.com/FrederikLizakJohansen/DebyeCalculator/blob/main/Tutorial.ipynb), which is a widely recognised software to use in the field. 2) to facilitate an easy way of simulating scattering patterns from many structures with the same structural parameters. Experimental data comes with one set of parameters like Q-range, Q-step, radiation-type etc. and therefore, we often want to fix the parameters in the simulation to those of the experiment and simulate the scattering pattern of many structures to compare with the experiment.

With this implementation, the _many structures_ can also easily be variants of a loaded structure without initialising the 'DebyeCalculator' again. For example:
```python
# Initialise calculator object
calc = DebyeCalculator(qmin=1.0, qmax=8.0, qstep=0.01)

# Define structure sources
structure_tuple = (
    ["Fe", "Fe", "O", "O"],
    torch.tensor(
        [[0.5377, 0.7068, 0.8589],
         [0.1576, 0.1456, 0.8799],
         [0.5932, 0.0204, 0.6759],
         [0.6946, 0.4114, 0.4869]]
))

Q_fullStructure, I_fullStructure = calc.iq(structure_source=structure_tuple)
for i in range(len(structure_tuple[0])):
    structure_list = list(structure_tuple)
    del structure_list[0][i]
    structure_list[1] = structure_list[1][torch.arange(structure_list[1].size(0))!=i]
    structure_tuple = tuple(structure_list)

    Q_truncatedStructure, I_truncatedStructure = calc.iq(structure_source=structure_tuple)
```

We hope that this is a satisfactory reason for loading the structure at the method level. We will keep the issue open and label it 'idea for restructuring' for future inspiration.
